ast/misc.go

```diff
@@ -1144,19 +1144,18 @@ func (n *CreateUserStmt) Restore(ctx *RestoreCtx) error {
 		}
 	}
 
-	tslOptionLen := len(n.TslOptions)
+	tlsOptionLen := len(n.TLSOptions)
 
-	if tslOptionLen != 0 {
+	if tlsOptionLen != 0 {
 		ctx.WriteKeyWord(" REQUIRE ")
 	}
 
-	// Restore `tslOptions` reversely to keep order the same with original sql
-	for i := tslOptionLen; i > 0; i-- {
-		if i != tslOptionLen {
+	for i := 0; i < tlsOptionLen; i++ {
+		if i != 0 {
 			ctx.WriteKeyWord(" AND ")
 		}
-		if err := n.TslOptions[i-1].Restore(ctx); err != nil {
-			return errors.Annotatef(err, "An error occurred while restore CreateUserStmt.TslOptions[%d]", i)
+		if err := n.TLSOptions[i].Restore(ctx); err != nil {
+			return errors.Annotatef(err, "An error occurred while restore CreateUserStmt.TLSOptions[%d]", i)
 		}
 	}
```

Comment:
With normal iteration direction we could use a `for`/`range` loop instead of `for` loop.

```go
for i, option := range n.TLSOptions {
  if i != 0 {
    ctx.WriteKeyWord(" AND ")
  }
  if err := option.Restore(ctx); err != nil {
    return errors.Annotatef(err, "An error occurred while restore CreateUserStmt.TLSOptions[%d]", i)
  }
}
```